frontend/src/metabase/dashboard/components/DashboardGrid.jsx

```diff
@@ -224,10 +227,27 @@ export default class DashboardGrid extends Component {
     this.setState({ addSeriesModalDashCard: dc });
   }
 
+  getDashboardCardIcon = dashCard => {
+    const { isRegularCollection } = PLUGIN_COLLECTIONS;
+    const { dashboard } = this.props;
+    const isRegularQuestion = isRegularCollection({
+      authority_level: dashCard.collection_authority_level,
+    });
+    const isRegularDashboard = isRegularCollection({
+      authority_level: dashboard.collection_authority_level,
+    });
+    if (isRegularDashboard && !isRegularQuestion) {
+      const authorityLevel = dashCard.collection_authority_level;
+      const opts = PLUGIN_COLLECTIONS.AUTHORITY_LEVEL[authorityLevel];
+      return { name: opts.icon, color: color(opts.color), size: 14 };
+    }
+  };
```

Comment:
Maybe break this function down and have each function do more like one thing?

```suggestion
  checkIfShouldGetDashboardCardIcon = dashCard => {
    const { dashboard } = this.props;
    const { isRegularCollection } = PLUGIN_COLLECTIONS;
    const isRegularQuestion = isRegularCollection({
      authority_level: dashCard.collection_authority_level,
    });
    const isRegularDashboard = isRegularCollection({
      authority_level: dashboard.collection_authority_level,
    });

    return isRegularDashboard && !isRegularQuestion;
  };

  getDashboardCardIcon = dashCard => {
    const shouldGetDashboardCardIcon = this.checkIfShouldGetDashboardCardIcon(
      dashCard,
    );

    if (shouldGetDashboardCardIcon) {
      const authorityLevel = dashCard.collection_authority_level;
      const opts = PLUGIN_COLLECTIONS.AUTHORITY_LEVEL[authorityLevel];

      return { name: opts.icon, color: color(opts.color), size: 14 };
    }
  };

```

Reply:
I'd encourage keeping it straightforward and a bit coupled for now. Later we're also going to display "Verified" question icons (from the BUCM feature), so it will change anyway